File: school/summer2013/openmp/c/42_host_with_coprocessor.c
#include <stdio.h>

int main()
{
#pragma omp parallel
{
	#pragma omp sections
	{
		#pragma omp section
		{
			#pragma offload target(mic:0)
			#pragma omp parallel
			{
				printf("hello %d\n", omp_get_thread_num());
			}
		}
		#pragma omp section
		{
			#pragma offload target(mic:1)
			#pragma omp parallel
			{
				printf("hello %d\n", omp_get_thread_num());
			}
		}
	}
}
}

File: summer2013/openmp/fortran/42_host_with_coprocessor.f90
program host_with_coprocessor
    use omp_lib
    implicit none

!$omp parallel
    !$omp sections
    !$omp section
        !dir$ offload begin target(mic:0)
        !$omp parallel
            print *, 'hello ', omp_get_thread_num()
        !$omp end parallel
        !dir$ end offload

    !$omp section
        !dir$ offload begin target(mic:1)
        !$omp parallel
            print *, 'hello ', omp_get_thread_num()
        !$omp end parallel
        !dir$ end offload
    !$omp end sections
!$omp end parallel
end
